deployment/terraform/live/main.tf

```diff
@@ -9,18 +9,32 @@ terraform {
 
 provider "aws" {
   version = "~> 1.0"
-
-  region = "${var.region}"
+  region  = "${var.region}"
 }
 
 module "object_storage" {
   source           = "../modules/s3"
-  bucket_name_base = "${terraform.workspace}"
+  bucket_name_base = "${replace(terraform.workspace, "/[^A-Za-z0-9]/", "-")}"
 }
 
 module "identity_pool" {
   source                   = "../modules/cognito"
-  identity_pool_name       = "${var.identity_pool_name}"
+  identity_pool_name       = "${terraform.workspace}"
```

Comment:
Actually, it turns out that "Pool names can only contain numbers, letters, underscores and spaces", so this should be: `"${replace(terraform.workspace, "/[^A-Za-z0-9_ ]/", " ")}"`